cloud.foundry.cli/src/main/java/cloud/foundry/cli/logic/diff/DiffTreeCreator.java

```diff
@@ -0,0 +1,61 @@
+package cloud.foundry.cli.logic.diff;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import cloud.foundry.cli.logic.diff.change.CfChange;
+
+import javax.annotation.Nonnull;
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ * This class is responsible for creating a diff tree consisting of {@link DiffNode diff nodes} from a number of
+ * {@link CfChange change objects}.
+ */
+public class DiffTreeCreator {
+
+    /**
+     * Creates the a diff tree based on the given changes.
+     * Uses the path set in every change object to build the tree structure.
+     * @param changes list of changes
+     * @return DiffNode root node of the tree
+     * @throws NullPointerException when changes is null
+     */
+    public static DiffNode createFrom(List<CfChange> changes) {
+        checkNotNull(changes);
+
+        if (changes.size() > 0) {
+            // the first entry of every node is the root name
+            String rootName = changes.get(0).getPath().get(0);
+
+            DiffNode diffNode = new DiffNode(rootName);
+            for (CfChange change : changes) {
+                DiffTreeCreator.insert(diffNode, new LinkedList<>(change.getPath()), change);
+            }
+            return diffNode;
+        }
+        return new DiffNode("");
```

Comment:
Leaving it for now since it requires some other changes and doesn't break anything. Works as a Null Value Object Dummy and might be changed in the future.